### PR TITLE
feat: add inscription confirmation for sats connect

### PR DESCRIPTION
**Pull request by victorkirov**
# ⚠️ Note:
- [x] includes changes from #577 so that PR would be better merged in first for the diff to take it into account
- [ ] requires sats-connect and core PRs to be merged and deployed
- [x] ~~there is 1 todo left with the preview icon for HTML~~

# 🔘 PR Type
What kind of change does this PR introduce?

- [x] Enhancement

# 📜 Background
We want to be able to create inscriptions via our inscription service using Sats Connect. This adds the confirmation screens for the inscription creation.

Issue Link: ENG-2622

# 🔄 Changes
- adds the inpage, content and popup handling of the create inscription Sats Connect events
- adds a Review Inscription screen where the user can confirm their inscription details and set fee rates
- upgrades core to a version that has the functions to create inscriptions

Impact:
- This allows third party developers to create inscriptions via our inscription service by making use of the inscription functions in Sats Connect. It is best tested with the Sats Connect Example app with the inscription feature implemented.
 
# 🖼 Screenshot / 📹 Video
Include screenshots or a video demonstrating the changes. This is especially helpful for UI changes.

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "redux": "^4.0.5",
     "redux-persist": "^6.0.0",
     "redux-state-sync": "^3.1.4",
-    "sats-connect": "1.0.0",
+    "sats-connect": "^0.5.0-50cf653",
```
Comment (victorkirov) [inline package.json:55]: Note: pending deployment of the 2 deps

File: src/pages/Popup/index.tsx [hunk 2]
```diff
@@ -26,6 +26,10 @@ const renderApp = async () => {
     buster: VERSION,
   });
   const container = document.getElementById('app');
+
+  // We need to do this after load in order for the extension popup to have the correct height on first load
+  container!.style.maxHeight = '100vh';
+
```
Comment (teebszet) [inline src/pages/Popup/index.tsx:32]: @victorkirov this was part of another PR right? maybe we can add a note on the merge order to make this PR easier to review?

Comment (victorkirov) [reply src/pages/Popup/index.tsx:32]: Ah, yes. Sorry, will add that 👍

Comment (teebszet) [reply src/pages/Popup/index.tsx:32]: thanks! actually, I guess it doesn't help a whole lot 😂

File: src/common/types/inpage-types.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@ export enum DomEventName {
   signPsbtRequest = 'SatsPsbtRequest',
   signMessageRequest = 'SatsSignMessage',
   sendBtcRequest = 'SatsSendBtcRequest',
+  createInscriptionRequest = 'createInscriptionRequest',
```
Comment (abdulhaseeb4239) [inline src/common/types/inpage-types.ts:13]: why did we not follow the pattern and assign 'SatsCreateInscriptionRequest'

Comment (victorkirov) [reply src/common/types/inpage-types.ts:13]: Ah, good catch. I didn't notice the pattern. Thanks 😄

File: src/app/screens/createInscription/index.tsx [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
+                      value={commitValue}
+                      displayType="text"
+                      thousandSeparator
+                      suffix=" sats"
+                    />
+                    <NumericFormat
+                      value={fiatValue}
+                      displayType="text"
+                      thousandSeparator
+                      suffix=" USD"
```
Comment (abdulhaseeb4239) [inline src/app/screens/createInscription/index.tsx:368]: This should not be hardcoded I think, but rather picked up from selected fiat currency. 
`const { fiatCurrency } = useWalletSelector();`

File: src/app/screens/createInscription/index.tsx [hunk 5]
```diff
@@ -0,0 +1,479 @@
...
+                        value={totalFee}
+                        displayType="text"
+                        thousandSeparator
+                        suffix=" sats"
+                      />
+                      <NumericFormat
+                        value={fiatFees}
+                        displayType="text"
+                        thousandSeparator
+                        suffix=" USD"
```
Comment (abdulhaseeb4239) [inline src/app/screens/createInscription/index.tsx:436]: same here, selected fiat currency should be rendered

**Review by abdulhaseeb4239 — CHANGES_REQUESTED** [review]

File: src/app/screens/createInscription/index.tsx (hunk 4 above)
Comment (victorkirov) [reply src/app/screens/createInscription/index.tsx:368]: Ah, woops! Thanks for catching this 🙏

File: src/app/screens/createInscription/index.tsx [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+                </CardRow>
+              </>
+            )}
+          </CardContainer>
+          <Button onClick={onAdvancedSettingClick}>
+            <ButtonImage src={SettingIcon} />
+            <ButtonText>{t('EDIT_FEES')}</ButtonText>
+          </Button>
+          {showFeeSettings && (
+            <EditFee
```
Comment (abdulhaseeb4239) [inline src/app/screens/createInscription/index.tsx:459]: why can't we use the `TransactionSettingAlert` for the Edit fee. as it is used in other transaction screens?

**Review by abdulhaseeb4239 — DISMISSED** [review]

File: src/app/screens/createInscription/index.tsx (hunk 6 above)
Comment (victorkirov) [reply src/app/screens/createInscription/index.tsx:459]: That component handles so many different scenarios like BTC payment, ordinal payment, stacks payment. I didn't want to add another method in there and confuse the logic even further. That component needs a rewrite to separate all those into separate components, so I am using a custom local edit fees component for now.

File: src/app/screens/createInscription/ContentLabel/index.tsx [hunk 7]
```diff
@@ -0,0 +1,236 @@
+import axios from 'axios';
+import { MouseEvent as ReactMouseEvent, useEffect, useMemo, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import styled from 'styled-components';
+
+import DotsVerticalIcon from '@assets/img/dots_three_vertical.svg';
+import EyeIcon from '@assets/img/eye.svg';
+import ShareIcon from '@assets/img/share.svg';
```
Comment (teebszet) [inline src/app/screens/createInscription/ContentLabel/index.tsx:8]: FYI we are in the middle of migrating to phosphor icon set, see here for usage:
https://github.com/secretkeylabs/xverse-web-extension/blob/069859a25e37c2dbbaabc93a151be1dcc45189fc/src/app/components/seedPhraseInput/index.tsx#L69

Comment (victorkirov) [reply src/app/screens/createInscription/ContentLabel/index.tsx:8]: 🙌  That's awesome. Will update it now.

**Review by yknl — APPROVED** [review]

**Pull request merged**